src/player_display.cpp

```diff
@@ -765,460 +878,560 @@ Strength - 4;    Dexterity - 4;    Intelligence - 4;    Perception - 4" ) );
-                } else if( line >= ( bionicslist.size() - ( bionics_useful_size_y + 1 ) / 2 ) ) {
-                    min = ( bionicslist.size() < bionics_useful_size_y ? 0 : bionicslist.size() - bionics_useful_size_y
-                            + 1 );
-                    max = bionicslist.size();
-                } else {
-                    min = line - ( bionics_useful_size_y - 1 ) / 2;
-                    max = std::min( bionicslist.size(), ( size_t )( 1 + line + bionics_useful_size_y / 2 ) );
-                }
-
-                for( size_t i = min; i < max; i++ ) {
-                    trim_and_print( w_bionics, int( 2 + i - min ), 1, getmaxx( w_bionics ) - 1,
-                                    i == line ? hilite( c_white ) : c_white, bionicslist[i].info().name );
-                }
-                if( line < bionicslist.size() ) {
-                    fold_and_print( w_info, 0, 1, FULL_SCREEN_WIDTH - 2, c_white,
-                                    bionicslist[line].info().description );
-                }
-                wrefresh( w_bionics );
-                wrefresh( w_info );
-
-                action = ctxt.handle_input();
-                if( action == "DOWN" ) {
-                    if( line < bionicslist.size() - 1 ) {
-                        line++;
-                    }
-                    break;
-                } else if( action == "UP" ) {
-                    if( line > 0 ) {
-                        line--;
-                    }
-                } else if( action == "NEXT_TAB" ) {
-                    mvwprintz( w_bionics, 0, 0, c_light_gray, header_spaces.c_str() );
-                    center_print( w_bionics, 0, c_light_gray, title_BIONICS );
-                    trim_and_print( w_bionics, 1, 1, getmaxx( w_bionics ) - 1, c_white,
-                                    string_format( _( "Bionic Power: %1$d" ), max_power_level ) );
-                    for( size_t i = 0; i < bionicslist.size() && i < bionics_win_size_y; i++ ) {
-                        mvwprintz( w_bionics, int( i + 2 ), 1, c_black, "                         " );
-                        trim_and_print( w_bionics, int( i + 2 ), 1, getmaxx( w_bionics ) - 1,
-                                        c_white, bionicslist[i].info().name );
-                    }
-                    wrefresh( w_bionics );
-                    line = 0;
-                    curtab++;
-                } else if( action == "QUIT" ) {
-                    done = true;
-                }
-                break;
+        if( aSkill->ident() == skill_id( "dodge" ) ) {
+            mvwprintz( w_skills, int( 1 + i - min ), 14, cstatus, "%4.1f/%-2d(%2d%%)",
+                       player.get_dodge(), level.level(), exercise < 0 ? 0 : exercise );
+        } else {
+            mvwprintz( w_skills, int( 1 + i - min ), 19, cstatus, "%-2d(%2d%%)", level.level(),
+                       ( exercise <  0 ? 0 : exercise ) );
```

Comment:
There are several places with redundant parentheses. But they are also unnecessary in expressions like `( a > b ) ? x : y`.